File: plugin/src/main/java/ru/arc/ai/tools/ToolMessage.java
package ru.arc.ai.tools;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Builder;
import lombok.Data;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

@Data
@Builder
public class ToolMessage {
    public UUID uuid;
    public String toolName;
    public Tool toolDto;
    public long timestamp;

    transient public CompletableFuture<ToolResult> responseFuture;
    @Builder.Default
    transient public ToolResult toolResult = new ToolResult();
    transient public boolean atLeastOneResponse;
}

File: src/main/kotlin/ru/arc/ai/tools/ToolMessage.kt
package ru.arc.ai.tools

import java.util.UUID
import java.util.concurrent.CompletableFuture

data class ToolMessage(
    @JvmField val uuid: UUID,
    @JvmField val toolName: String,
    @JvmField val toolDto: Tool,
    @JvmField val timestamp: Long,
    @JvmField @Transient var responseFuture: CompletableFuture<ToolResult>,
    @JvmField @Transient var toolResult: ToolResult = ToolResult(),
    @JvmField @Transient var atLeastOneResponse: Boolean = false,
)
